src/meta/processors/job/BalanceJobExecutor.h

```diff
@@ -32,9 +34,199 @@ class BalanceJobExecutor : public MetaJobExecutor {
+
+ public:
+  DataBalanceJobExecutor(JobDescription jobDescription,
+                         kvstore::KVStore* kvstore,
+                         AdminClient* adminClient,
+                         const std::vector<std::string>& params)
+      : BalanceJobExecutor(jobDescription.getJobId(), kvstore, adminClient, params),
+        jobDescription_(jobDescription) {}
+  nebula::cpp2::ErrorCode recovery() override;
+  nebula::cpp2::ErrorCode prepare() override;
+  nebula::cpp2::ErrorCode finish(bool ret = true) override;
+  nebula::cpp2::ErrorCode stop() override;
+
+ protected:
+  folly::Future<Status> executeInternal(HostAddr&& address,
+                                        std::vector<PartitionID>&& parts) override;
+  nebula::cpp2::ErrorCode buildBalancePlan();
+  ErrorOr<nebula::cpp2::ErrorCode, std::vector<BalanceTask>> genTasks(
+      GraphSpaceID spaceId,
+      int32_t spaceReplica,
+      bool dependentOnGroup,
+      std::vector<HostAddr>& lostHosts);
+  ErrorOr<nebula::cpp2::ErrorCode, HostAddr> hostWithMinimalParts(const HostParts& hostParts,
+                                                                  PartitionID partId);
+
+  ErrorOr<nebula::cpp2::ErrorCode, HostAddr> hostWithMinimalPartsForZone(const HostAddr& source,
+                                                                         const HostParts& hostParts,
+                                                                         PartitionID partId);
+  bool balanceParts(GraphSpaceID spaceId,
+                    HostParts& confirmedHostParts,
+                    int32_t totalParts,
+                    std::vector<BalanceTask>& tasks,
+                    bool dependentOnGroup);
+  ErrorOr<nebula::cpp2::ErrorCode, std::pair<HostParts, std::vector<HostAddr>>> fetchHostParts(
+      GraphSpaceID spaceId,
+      bool dependentOnGroup,
+      const HostParts& hostParts,
+      std::vector<HostAddr>& lostHosts);
+  nebula::cpp2::ErrorCode transferLostHost(std::vector<BalanceTask>& tasks,
+                                           HostParts& confirmedHostParts,
+                                           const HostAddr& source,
+                                           GraphSpaceID spaceId,
+                                           PartitionID partId,
+                                           bool dependentOnGroup);
+  Status checkReplica(const HostParts& hostParts,
+                      const std::vector<HostAddr>& activeHosts,
+                      int32_t replica,
+                      PartitionID partId);
+  std::vector<std::pair<HostAddr, int32_t>> sortedHostsByParts(const HostParts& hostParts);
+  bool checkZoneLegal(const HostAddr& source, const HostAddr& target);
+  nebula::cpp2::ErrorCode finishInternal(bool ret = true);
+
+ private:
+  static std::unique_ptr<BalancePlan> plan_;
+  std::vector<HostAddr> lostHosts_;
+  JobDescription jobDescription_;
+};
+
+class LeaderBalanceJobExecutor : public BalanceJobExecutor {
```

Comment:
`balance leader` just keep it in Balancer will be ok? I think it is not a job/task actually.

Since both old and new commands are reserved. `balance leader` will impact all spaces, but `balance data` will only work on specific space.

Just we only need `DataBalanceJobExecutor` instead of inheriting from `BalanceJobExecutor`. WDYT?